### PR TITLE
ADBDEV-4377: Fix backup statistics for leaf partitions

### PR DESCRIPTION
**Pull request by RekGRpth**
Fix backup statistics for leaf partitions

gpbackup does not backup statistics for leaf partitions with leaf-partition-data option

This patch solves the problem by backuping statistics for all specified
tables, including leaf partitions, and not just root partitions.

The test for this patch is based on the test for the 85384fadf774560b87df4d30792644b87069583e.
And, as stated in that patch, it removes the unnecessary
condition when comparing the number of statistics.

### CODE REVIEW

File: backup/wrappers.go [hunk 1]
```diff
@@ -220,7 +220,7 @@ func RetrieveAndProcessTables() ([]Table, []Table) {
 	metadataTables, dataTables := SplitTablesByPartitionType(tables, quotedIncludeRelations)
 	objectCounts["Tables"] = len(metadataTables)
 
-	return metadataTables, dataTables
+	return metadataTables, dataTables, tables
```
Comment (red1452) [inline backup/wrappers.go:223]: Rename `tables` to `allTables`.

**Review by red1452 — COMMENTED** [review]
You should change branch `ADBDEV-4376` to `master`.

**Comment by RekGRpth** [comment]
> You should change branch `ADBDEV-4376` to `master`.

Done

File: end_to_end/end_to_end_suite_test.go [hunk 2]
```diff
@@ -1485,13 +1485,6 @@ var _ = Describe("backup and restore end to end tests", func() {
 			skipIfOldBackupVersionBefore("1.18.0")
 
 			testhelper.AssertQueryRuns(backupConn, `
-				CREATE TABLE et (
-					id character varying(13),
-					flg smallint,
-					dttm timestamp without time zone,
-					src character varying(80)
-				) WITH (appendonly='true', orientation='row', compresstype=zstd, compresslevel='3') DISTRIBUTED BY (id);
-
```
[the remark this answers is not in the document]
Comment (RekGRpth) [reply end_to_end/end_to_end_suite_test.go:1494]: It is no longer needed.

File: backup/wrappers.go (hunk 1 above)
Comment (RekGRpth) [reply backup/wrappers.go]: Done

File: end_to_end/end_to_end_suite_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (red1452) [reply end_to_end/end_to_end_suite_test.go:1494]: How do these changes relate to the problem?

[the remark this answers is not in the document]
Comment (RekGRpth) [reply end_to_end/end_to_end_suite_test.go:1494]: This table was added by a previous commit to reproduce the bug, but with this commit, that bug is reproduced without the additional table.

**Comment by red1452** [comment]
I think, you should rewrite the head of commit message. Head should answer the question: "what had been done?". For example: "Fix something at somewhere".

**Comment by RekGRpth** [comment]
> you should rewrite the head of commit message

I changed the name and content of the PR. Will this work for a commit? If so, can I force push changes?

**Comment by red1452** [comment]
> I changed the name and content of the PR. Will this work for a commit? If so, can I force push changes?

You can add empty commit with new commit message without force-push.

**Review by red1452 — APPROVED** [review]

**Review by dnskvlnk — APPROVED** [review]

**Pull request merged**